internal/app/admin/http/handler_test.go

```diff
@@ -337,6 +341,145 @@ func TestAdminServer_CacheDumpHandler_EntireCache(t *testing.T) {
+		upstreamResponseChannelLDS := make(chan *discoveryv3.DiscoveryResponse)
+		upstreamResponseChannelCDS := make(chan *discoveryv3.DiscoveryResponse)
+		mockScope := tally.NewTestScope("mock_orchestrator", make(map[string]string))
+		client := upstream.NewMockV3(
+			ctx,
+			upstream.CallOptions{Timeout: time.Second},
+			nil,
+			upstreamResponseChannelLDS,
+			nil,
+			nil,
+			upstreamResponseChannelCDS,
+			func(m interface{}) error { return nil },
+		)
+		orchestrator := orchestrator.NewMock(t, mapper, client, mockScope)
+		assert.NotNil(t, orchestrator)
+
+		gcpReq1 := gcpv3.Request{
+			TypeUrl: resourcev3.ListenerType,
+		}
+		ldsRespChannel, cancelLDSWatch := orchestrator.CreateWatch(transport.NewRequestV3(&gcpReq1))
+		assert.NotNil(t, ldsRespChannel)
+
+		gcpReq2 := gcpv3.Request{
+			TypeUrl: resourcev3.ClusterType,
+		}
+		cdsRespChannel, cancelCDSWatch := orchestrator.CreateWatch(transport.NewRequestV3(&gcpReq2))
+		assert.NotNil(t, cdsRespChannel)
+
+		listener := &v2.Listener{
+			Name: "lds resource",
+		}
+		listenerAny, err := ptypes.MarshalAny(listener)
+		assert.NoError(t, err)
+		resp := discoveryv3.DiscoveryResponse{
+			VersionInfo: "1",
+			TypeUrl:     resourcev3.ListenerType,
+			Resources: []*any.Any{
+				listenerAny,
+			},
+		}
+		upstreamResponseChannelLDS <- &resp
+		gotResponse := <-ldsRespChannel.GetChannel().V3
+		gotDiscoveryResponse, err := gotResponse.GetDiscoveryResponse()
+		assert.NoError(t, err)
+		assert.Equal(t, resp, *gotDiscoveryResponse)
+
+		cluster := &v2.Cluster{
+			Name: "cds resource",
+		}
+		clusterAny, err := ptypes.MarshalAny(cluster)
+		assert.NoError(t, err)
+		resp = discoveryv3.DiscoveryResponse{
+			VersionInfo: "2",
+			TypeUrl:     resourcev3.ClusterType,
+			Resources: []*any.Any{
+				clusterAny,
+			},
+		}
+		upstreamResponseChannelCDS <- &resp
+		gotResponse = <-cdsRespChannel.GetChannel().V3
+		gotDiscoveryResponse, err = gotResponse.GetDiscoveryResponse()
+		assert.NoError(t, err)
+		assert.Equal(t, resp, *gotDiscoveryResponse)
+
+		req, err := http.NewRequest("GET", url, nil)
+		assert.NoError(t, err)
+
+		rr := httptest.NewRecorder()
+		handler := cacheDumpHandler(&orchestrator)
+
+		handler.ServeHTTP(rr, req)
+		assert.Equal(t, http.StatusOK, rr.Code)
+		assert.Contains(t, rr.Body.String(), `ldsv3: {
+  "Resp": {
+    "VersionInfo": "1",
+    "Resources": {
+      "Endpoints": null,
+      "Clusters": null,
+      "Routes": null,
+      "Listeners": [
+        {
+          "name": "lds resource"
+        }
+      ],
+      "Secrets": null,
+      "Runtimes": null,
+      "Unmarshalled": null
+    },
+    "Canary": false,
+    "TypeURL": "type.googleapis.com/envoy.config.listener.v3.Listener",
+    "Nonce": "",
+    "ControlPlane": null
+  },
+  "Requests": [
+    {
+      "version_info": "1",
+      "type_url": "type.googleapis.com/envoy.config.listener.v3.Listener"
+    }
+  ],
+  "ExpirationTime": "`)
+		assert.Contains(t, rr.Body.String(), `cdsv3: {
+  "Resp": {
```

Comment:
Can we add these a fixtures in testdata? For the expirationTime, you can probably use Go templates.

Reply:
separated the fixtures